modules/plugin/mongodb/src/main/java/org/geotools/data/mongodb/MongoUtil.java

```diff
@@ -167,4 +174,45 @@ private static void setDBOValueInternal(
         }
         return null;
     }
+
+    public static String extractFilesNameFromUrl(String url) {
+        return url.substring(url.lastIndexOf('/') + 1, url.length());
```

Comment:
A detail here, seems like we are building the cache file name from the last path-separator (that is good of course) but er are not deleting the query character (?) and query parameters potentially included in a HTTP resource path, generating filenames like "filename.zip?param=1&query=44".  Also we could lead with a resource from a rest service ended with a path-separator.  It would be nice to handle these scenarios as well.

Reply:
understood, however if you can list down the scenarios of possible url combinations, it would be easier to design a unit test

Reply:
Here I would follow another approach, you get a ZIP file, and you interested only in its content, the schemas data directory providers I described above are the ones responsible to provide the location to store them, we got a ZIP or a file, we clear the current schemas directory content and drop the ZIP output or file in it.

When handling the ZIP file, it may happens that the user ZIP the schemas as root or a directory, we shoudl vouch for the two use cases:
1. we have a directory inside a ZIP, we use the directory content
2. we have files as root inside the directory, we use whatever is there